### PR TITLE
Binderator improvements

### PR DESCRIPTION
**Pull request by moljac**
Adding some trivial improvements to `binderator`:

- download additional files for bindings improvements

  POM files, JavaSources and JavaDocs (for parameter names)

- batch operations

- tracing 

## batch operations

`binderator` would exit after 1st unsatisfied dependency which was productivity killer for large bindings like AndroidX and Google.Play.Services-Firebase-MLKit.

With this improvement `binderator` continues checking dependencies for all artifacts in `config.json` file and throws exception for all dependencies:

Example:

```
Dependency errors : 2
1
	No matching artifact config found for: org.jetbrains.kotlin.kotlin-stdlib:1.4.32 to satisfy dependency of: androidx.window.window:1.0.0-alpha06
2
	No matching artifact config found for: org.jetbrains.kotlinx.kotlinx-coroutines-android:1.4.3 to satisfy dependency of: androidx.window.window:1.0.0-alpha06

   at AndroidBinderator.Engine.BuildProjectModels(BindingConfig config, TemplateConfig template, Dictionary`2 mavenProjects) in /Users/Shared/Projects/d/X/XC/binderator-improvements/Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs:line 386
   at AndroidBinderator.Engine.ProcessConfig(MavenRepository maven, BindingConfig config) in /Users/Shared/Projects/d/X/XC/binderator-improvements/Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs:line 86
   at AndroidBinderator.Engine.BinderateAsync(BindingConfig config) in /Users/Shared/Projects/d/X/XC/binderator-improvements/Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs:line 56
   at Xamarin.AndroidBinderator.Tool.Program.Main(String[] args) in /Users/Shared/Projects/d/X/XC/binderator-improvements/Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator.Tool/Program.cs:line 64
   at Xamarin.AndroidBinderator.Tool.Program.<Main>(String[] args)
```

### CODE REVIEW

File: Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs [hunk 1]
```diff
@@ -311,8 +370,25 @@ static List<BindingProjectModel> BuildProjectModels(BindingConfig config, Templa
 						}
 					});
 				}
+				if (exceptions.Any())
+                {
```
Comment (jonpryor) [inline Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs:374]: indentation needs fixing.

File: Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs [hunk 2]
```diff
@@ -311,8 +370,25 @@ static List<BindingProjectModel> BuildProjectModels(BindingConfig config, Templa
...
+					StringBuilder sb = new StringBuilder();
```
Comment (jonpryor) [inline Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs]: "More conventional" approach would be to instead do:

```csharp
throw new AggregateException(exceptions);
```

File: Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs (hunk 1 above)
Comment (moljac) [reply Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs:374]: fixed

Comment (moljac) [reply Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs:374]: added editorconfig from `xamarin-android` repo

File: Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs (hunk 2 above)
Comment (moljac) [reply Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs]: fixed and output (exceptions dump) moved to console app (Main in Program.cs)

**Review by jpobst — APPROVED** [review]

**Pull request merged**